### DISCUSSION TITLE
How to use CommunityToolkit with VB?

### DISCUSSION THREAD

**Discussion by DrCQ** [CommunityToolkit/dotnet#565] [Q&A] [answered]
Hi,
I use now the version 8.0 of the CommunityToolkit with VB.NET in the classic way - means: deriving classes from the ObservableObject etc. I would like to use the full power of the package, including the Attributes for properties and commands, which are generating own code.
At the moment, I was able to use the validation attributes like <RegularExpression()> and own class derived from the ValidationAttribute class. It is a way to "enable" VB code and use Roslyn to generate VB code? Or it is planned to add support for VB?

**Comment by Sergio0694** [comment, marked as answer]
Hey there! I'm afraid there's no VB support available for our generators, nor planned 😅
All our generators are specifically for C#, and we will not be adding support for other languages. To be clear, our generators also specifically need some C# features that are not available in an equivalent way on VB, so it wouldn't really be doable to extend support there as well.

**Reply by Brains** [reply]
> our generators also specifically need some C# features that are not available in an equivalent way on VB

What about `F#`? It lacks those features as well? 
However, if you don't use it, you might not know about all available features. Will try to investigate that by myself.
Can you provide the list of those `C#` features, or where I can find it?
